### ISSUE TITLE
Add DIAMOND/BWA support

### ISSUE THREAD

**Issue by wwood** [wurmlab/sequenceserver#134]
DIAMOND is a fast blastx/blastp replacement
http://www.nature.com/nmeth/journal/vaop/ncurrent/full/nmeth.3176.html

Maybe could output using `--sam` and then convert that to HTML with the reference and query sequences in some accessible (`faidx`?) format somehow? If this was done then using `bwa` instead of blastn would presumably be not much different.

Like #128, this is probably low priority.

**Comment by yannickwurm** [comment]
yeah I had a look into this - unfortunately there is no plan for xml-type output; only a blast-table like if I remember correctly...

**Comment by wwood** [comment]
Presumably sam would be better than blast-table since the cigar string gives the actual alignment, but I agree the XML would have made things easier.

**Comment by jffkm** [comment]
I had a chance to performance test diamond and it's ridiculously fast for large numbers of queries. For single queries, blastp is about equivalent. This would be a great feature for drag-n-dropping a file with lots of fasta entries though.

**Comment by jesalo** [comment]
DIAMOND has had a lot of development recently, and is a tool that can produce BLAST compatible outputs. It is multiple times faster than BLAST - would you reconsider enabling Diamond for SequenceServer?

**Comment by yeban** [comment]
XML output generated by DIAMOND can now be imported and visualised in SequenceServer.

I am closing this issue. Please open a new one for BWA support if you still think is relevant.

Thanks.

**Comment by Rridley7** [comment]
Hello, I wanted to see if there were any updates on this feature? Looking to use a local seqserv instance with a large protein database, and this would be very useful. Thanks for your work on this tool!

**Comment by lukaszsobala** [comment]
Since version 2.0.9, DIAMOND can use BLAST databases: https://github.com/bbuchfink/diamond/releases/tag/v2.0.9 (with a small preparation step added in the next version: https://github.com/bbuchfink/diamond/releases/tag/v2.0.10)
So it could be a faster, drop-in replacement for the BLAST binaries. Just the variables affecting memory usage will have to be tuned.
